### PR TITLE
Adding database_retention field support to Cloud Composer

### PR DESCRIPTION
**Pull request by apilaskowski**
Add database_retention field support to Cloud Composer

If this PR is for Terraform, I acknowledge that I have:

- [x]  Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x]  Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x]  [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [make test and make lint](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x]  [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x]  Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.
- [x] Release Note Template for Downstream PRs (will be copied)

Release Note Template for Downstream PRs (will be copied)
```release-note:enhancement
composer: added `data_retention_config.airflow_metadata_retntion_config` field on `google_composer_environment`
```
Fixes https://github.com/hashicorp/terraform-provider-google/issues/17898

### CODE REVIEW

**Comment by apilaskowski** [comment]
I know what is the problem, I need to consult and solve that.

**Comment by apilaskowski** [comment]
I will update the doc ASAP.

**Comment by apilaskowski** [comment]
Hi @BBBmau,

I think you can review it now.

Please let me know if you think sth shoutl be updated.

File: mmv1/third_party/terraform/services/composer/resource_composer_environment.go.erb [hunk 1]
```diff
@@ -746,18 +751,46 @@ func ResourceComposerEnvironment() *schema.Resource {
...
-										Required:     true,
+										Optional:     true,
+										ForceNew:     false,
+										AtLeastOneOf: composerDataRetentionKeys,
 										Elem: &schema.Resource{
 											Schema: map[string]*schema.Schema{
 												"storage_mode": {
 													Type:         schema.TypeString,
 													Optional:     true,
+													ForceNew:     false,
```
Comment (BBBmau) [inline mmv1/third_party/terraform/services/composer/resource_composer_environment.go.erb:762]: Seems like the force new is happening due to the resource using the default value from the api perhaps?

```
                 ~ data_retention_config {
                      ~ airflow_metadata_retention_config {
                          - retention_days = 100 -> null
                          ~ retention_mode = "RETENTION_MODE_ENABLED" -> "RETENTION_MODE_DISABLED"
                        }
        
                        # (1 unchanged block hidden)
                    }
```

Comment (apilaskowski) [reply mmv1/third_party/terraform/services/composer/resource_composer_environment.go.erb:762]: I have more or less the idea what is wrong here. I will notice you asap when I update. 
Sorry for the delay.

Creations are working properly.
Updates have a bug.

**Comment by apilaskowski** [comment]
I have this item on my TODO list, and I am getting there.
According to our predictions, this item should be ready around end of the June, beginning of July.

**Comment by c2thorn** [comment]
Hi @apilaskowski 

Tomorrow, the Magic Modules repository is scheduled to undergo a language migration from Ruby to Go. You can view more details about this in our announcement here: https://github.com/hashicorp/terraform-provider-google/issues/19583 (or [go/mm-migration-announcement](go/mm-migration-announcement) if you are a Googler)

This open pull request may become incompatible due to most YAML and .erb files converting to Go-compatible files.

Our team (Magic Modules repository maintainers) has tooling to automatically convert changes to the new language, and we can prepare a new commit for this pull request that is compatible with the migration.

In order to push the new changes to your pull request, we would need to <ins>force push the commit to your fork's branch</ins>. Our tooling saves a backup branch before converting, so we could rollback or open a new pull request if needed. We would also work with you and the PR reviewer in the event additional changes are needed.

You also have the option to update the pull request yourself after the migration. You can view a [preview branch](https://github.com/GoogleCloudPlatform/magic-modules/tree/FEATURE-BRANCH-rewrite-switchover-preview) and [updated documentation](https://googlecloudplatform.github.io/magic-modules/reference/ruby-go-changes/) related to the migration changes.

We will take no action until we have your explicit permission to push changes to your fork's branch used for this pull request. Let me or your reviewer know if you have any further questions!

**Comment by c2thorn** [comment]
It has been several weeks and this PR is likely too far out of date to convert with our tooling. Please disregard my previous comment and recreate this PR in the new format.

**Pull request closed without merging**